<a href="https://colab.research.google.com/github/baileymccluresinger/PUS2024_BSinger/blob/main/HW3/HW3_Assignment1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**HW3: NYC Post-Prison Employment Programs**
---

In [2]:
import os
import sys
import numpy as np
import pylab as pl

%pylab inline

from IPython.display import Image
#Image(filename='https://www.mdrc.org/sites/default/files/What%20Strategies%20Work%20for%20the%20Hard%20FR.pdf')

Populating the interactive namespace from numpy and matplotlib


##**Null Hypothesis:** the % of former prisoners who are convicted with a felony after release is the same or higher for candidates who participated in the program as for the control group, significance level $\alpha$ $=$ $0.05$

### _$H_0$_ : $\\{P_{\\\\{0}}}$ $-$ $\\{P_{\\\\{1}}}$ $<=$ $0$

### _$H_A$_ : $\\{P_{\\\\{0}}}$ $-$ $\\{P_{\\\\{1}}}$ $>$ $0$

### $\alpha$ $=$ $0.05$

### $\\{P_{\\\\{0}}}$ $=$ $0.117$ $,$ $\\{P_{\\\\{1}}}$ $=$ $0.1$

In [75]:
#significance threshold
alpha=0.05


# 0 for control group, 1 for program group

P_0 = 11.7 * 0.01
P_1 = 10 * 0.01

if P_0 - P_1 >= 0:
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

#sample sizes

n_0 = 409
n_1 = 568

#count calculations

Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

the Null holds


In [76]:
#define the sample proportion
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.10711668372569089


In [77]:
#calculate p
p = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)
#standard error
se = lambda p, n0, n1: np.sqrt(p * (1 - p) * (1.0 / n0 + 1.0 / n1))

In [78]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_2y = zscore(P_1, P_0, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
print (z_2y)

-0.8476402052199316


In [59]:
#p-value = 1 - P

p_2y = 1 - 0.8389


def report_result(p,a):
    print ('is the p value ' +
           '{0:.2f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else:
        print ("NO!")

    print ('the Null hypothesis is {}'.format(\
                            'rejected' if p < a  else 'not rejected') )


report_result(p_2y, alpha)

is the p value 0.16 smaller than the critical value 0.05?
NO!
the Null hypothesis is not rejected


In [8]:
#felony conviction|    no     |    yes   |total|
#-----------------|-----------|----------|-----|
#test sample      |  558/568  |  10/568  | 568 |
#control sample   | 397.3/409 | 11.7/409 | 409 |
#total            |   955.3   |   21.7   | 977 |

In [79]:
#calculating expected values for each cell
cellone = (955.3*568)/977 #test and no
celltwo = (955.3*409)/977 #control and no
cellthree = (21.7*568)/977 #test and yes
cellfour = (21.7*409)/977 #control and yes
expected_values = np.array([[cellone, cellthree],[celltwo, cellfour]])
expected_values

array([[555.38423746,  12.61576254],
       [399.91576254,   9.08423746]])

In [74]:
#simple chi-square function
observed_values = np.array([[(558/568)*568, (10/568)*568],[(397.3/409)*409, (11.7/409)*409]])
#this is the same as -> observed_values = np.array([[558, 10],[397.3, 11.7]])

chi_squared_stat = (((observed_values-expected_values)**2)/expected_values).sum()

chi_squared_stat

1.3249795255620942

In [71]:
#external chi-square function
def evalChisq(values):
    values = np.array(values)
    E = np.empty_like(values)
    for j in range(len(values[0])):
        for i in range(2):

            E[i][j] = ((values[i,:].sum() * values[:,j].sum()) /
                        (values).sum())
    return ((values - E)**2 / E).sum()

In [73]:
#second check of chi-square value
Ntot = 977

sample_values = np.array([[(558/568)*568, (10/568)*568],[(397.3/409)*409, (11.7/409)*409]])

print(evalChisq(sample_values))

1.3249795255620942


###**Assignment 1 Questions**
---
#### **1.** The chi-square critical value, 3.84, is determined based on the singular degree of freedom and the significance threshold, $\alpha=0.05$.

#### **2.** The chi-square statistic that I determined, 1.32, is less than 3.84, which means the null hypothesis is rejected because the data deviated significantly from the expected distribution.

#### **3.**